deepspeed/runtime/engine.py

```diff
@@ -2885,7 +2879,12 @@ def _checkpoint_tag_validation(self, tag):
             elif not valid:
                 logger.warning(msg)
 
-    def save_checkpoint(self, save_dir, tag=None, client_state={}, save_latest=True):
+    def save_checkpoint(self,
+                        save_dir,
+                        tag=None,
+                        client_state={},
+                        save_latest=True,
+                        use_node_local_storage=False):
```

Comment:
Thanks for working on this. However, this node_local_storage option is better configured through ds_config since it is constant for a given run. Also, users can activate with ds_config change rather than code change.  
This option should specifically go [here](https://github.com/microsoft/DeepSpeed/blob/99fde3b7a573f2a261dc5ef9380870ce30a1db9d/deepspeed/runtime/constants.py#L368).